Ports/sqlite/package.sh

```diff
@@ -5,3 +5,4 @@ files="https://www.sqlite.org/2023/sqlite-autoconf-${version}.tar.gz sqlite-auto
 useconfigure='true'
 use_fresh_config_sub='true'
 workdir="sqlite-autoconf-${version}"
+depends=('zlib')
```

Comment:
Hm, I think `sqlite` built fine without `zlib` in the past, and I'm pretty sure it's an optional dependency?

The error without it is:

```
[sqlite/build] /bin/bash ./libtool  --tag=CC   --mode=link x86_64-pc-serenity-gcc -D_REENTRANT=1 -DSQLITE_THREADSAFE=1 -DSQLITE_ENABLE_MATH_FUNCTIONS -DSQLITE_ENABLE_FTS4 -DSQLITE_ENABLE_FTS5 -DSQLITE_ENABLE_RTREE -DSQLITE_ENABLE_GEOPOLY  -g -O2 -no-undefined -version-info 8:6:8  -o libsqlite3.la -rpath /usr/local/lib sqlite3.lo  
[sqlite/build] ./libtool: line 10196: shift: shift count out of range
[sqlite/build] libtool: link: x86_64-pc-serenity-gcc -shared  -fPIC -DPIC  .libs/sqlite3.o    -g -O2   -Wl,-soname -Wl, -o .libs/
[sqlite/build] /path/to/serenity/Toolchain/Local/x86_64/lib/gcc/x86_64-pc-serenity/13.1.0/../../../../x86_64-pc-serenity/bin/ld: SONAME must not be empty string; ignored
[sqlite/build] /path/to/serenity/Toolchain/Local/x86_64/lib/gcc/x86_64-pc-serenity/13.1.0/../../../../x86_64-pc-serenity/bin/ld: cannot open output file .libs/: Is a directory
[sqlite/build] collect2: error: ld returned 1 exit status
[sqlite/build] make: *** [Makefile:505: libsqlite3.la] Error 1
Error in step sqlite/build (status=2)
```

It seems like `libtool:10196` tries to shift out a library name from some variable, maybe that is failing because it doesn't need linking against any other libraries?

Reply:
This is caused by `library_names_spec` being empty, so it can't properly shift through the different combinations (#20365).

It indeed does seem like `zlib` isn't a required dependency, the package indeed builds properly after fixing up `libtool`.

Reply:
I misread a comment in the `configure` script that led me to believe `zlib` was required by the `sqlite3` shell. After building without `zlib` everything seems to work fine.